addons/azureSets/azuresets.lua

```diff
@@ -159,39 +199,42 @@ function find_spell_id_by_name(spellname)
     return nil
 end
 
-function set_single_spell(setspell,slot)
-    if windower.ffxi.get_player()['main_job_id'] ~= 16 --[[and windower.ffxi.get_player()['sub_job_id'] ~= 16]] then return nil end
+function set_single_spell(setspell, slot)
+    if not bluJobLevel then return nil end
 
-    local tmpTable = T(get_current_spellset())
-    for key,val in pairs(tmpTable) do
-        if tmpTable[key]:lower() == setspell then
+    update_current_spellset()
+    for key, val in pairs(currentSpellSet) do
+        if currentSpellSet[key]:lower() == setspell then
             error('That spell is already set.')
             return
         end
     end
     if tonumber(slot) < 10 then slot = '0'..slot end
     --insert spell add code here
-    for spell in spells:it() do
-        if spell['english']:lower() == setspell then
-            --This is where single spell setting code goes.
-            --Need to set by spell id rather than name.
-            windower.ffxi.set_blue_magic_spell(spell['id'], tonumber(slot))
+    local spellId = find_spell_id_by_name(setspell)
+    if spellId then
```

Comment:
You don't need this check, since you call `verify_and_set_spell` in the next step and that already checks for the validity of the spell ID.